core: add /input set_unread_current_buffer_here

mkoskar opened the pull request:
**Edited:** I've force-pushed patch with few cosmetic changes, especially in relation to #1007. 

* new "/input" action:

    set_unread_current_buffer_here: set unread marker at current position

* buffer "unread" property value now means:

    "0": set unread marker before first line
    int > 0: set unread marker after end line of current scroll position
    int < 0: set unread marker after last line of buffer

    Any other value is treated as "-1".

mkoskar commented:
This is partially fixing #552.

weechatter commented:
something i don't like. its too limited. i suggest adding the functionality from /window scroll

mkoskar commented:
@weechatter I was actually thinking about it, but then it seemed lot more work and maybe really needless duplication? I mean you can still use ``/window scroll`` separately before placing marker (but only on buffer that actually has window). Only then you don't have option to set marker before ``start_line`` of scroll (though that is much easier to implement than ``end_line``). I was basically catering for use-case where you can replace ``<Ctrl-S><Ctrl-U>`` binding and mark everything visible as read, be it after end of scroll or end of buffer. I thought it would be weird to place marker before ``start_line`` although it's something that could be add easily and then combined with ``/window scroll`` you can place it rather precisely, except only for "visible" buffers (that is having window). But I have a feeling you're suggesting it because you want to move it regardless of it being displayed in window or not. Hmm, well that's certainly bigger undertaking.

mkoskar committed:
core: add /input set_unread_current_buffer_here

* new "/input" action:

    set_unread_current_buffer_here: set unread marker at current position

* buffer "unread" property value now means:

    "0": set unread marker before first line
    int > 0: set unread marker after end line of current scroll position
    int < 0: set unread marker after last line of buffer

    Any other value is treated as "-1".
